spinalcordtoolbox/scripts/sct_get_centerline.py

```diff
@@ -151,7 +151,13 @@ def main(argv: Sequence[str]):
     contrast_type = arguments.c
     if method == 'optic' and not contrast_type:
         # Contrast must be
-        error = "ERROR: -c is a mandatory argument when using 'optic' method."
+        error = "ERROR: -c is a mandatory argument when using '-method optic'."
+        printv(error, type='error')
+        return
+
+    # Soft centerline option can only be used with fitseg method
+    if arguments.centerline_soft == 1 and method != 'fitseg':
+        error = "ERROR: -centerline-soft can only be used with '-method fitseg'."
         printv(error, type='error')
```

Comment:
Nitpick, but we can probably just combine these two lines into a single line?

```suggestion
        printv("ERROR: -centerline-soft can only be used with '-method fitseg'.", type='error')
```